dramatiq/results/backends/__init__.py

```diff
@@ -15,24 +15,8 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-import warnings
-
 from .stub import StubBackend
-
-try:
-    from .memcached import MemcachedBackend
-except ImportError:  # pragma: no cover
-    warnings.warn(
-        "MemcachedBackend is not available.  Run `pip install dramatiq[memcached]` "
-        "to add support for that backend.", ImportWarning,
-    )
-
-try:
-    from .redis import RedisBackend
-except ImportError:  # pragma: no cover
-    warnings.warn(
-        "RedisBackend is not available.  Run `pip install dramatiq[redis]` "
-        "to add support for that backend.", ImportWarning,
-    )
+from .memcached import MemcachedBackend
+from .redis import RedisBackend
 
 __all__ = ["StubBackend", "MemcachedBackend", "RedisBackend"]
```

Comment:
How about making imports lazy with a module-level `__getattr__`? That way, the changes to `memcached.py` and `redis.py` won't be necessary.  Something like this pseudocode:

```python
import importlib
import warnings

__all__ = ["StubBackend", "MemcachedBackend", "RedisBackend"]

def __getattr__(name):
  modname = {
    "MemcachedBackend": ".memcached",
    "RedisBackend": ".redis",
    "StubBackend": ".stub",
  }.get(name)
  if mod is None:
    raise AttributeError(f"module {__name__!r} has no attribute {name!r}")
  try:
    mod = importlib.import_module(modname, __name__)
    return getattr(mod, name)
  except ImportError:
    warnings.warn(...)
    raise
    


```

Reply:
I suppose you only support 3.7+ so that is an option. Fine by me if you're happy with this as a solution.

Reply:
Although at least due to your error messages, i might still prefer doing the checks non-dynamically; which is what i'll push up momentarily